apps/studio/pages/project/[ref]/auth/templates/[templateId].tsx

```diff
@@ -0,0 +1,114 @@
+import { PermissionAction } from '@supabase/shared-types/out/constants'
+
+import { useIsSecurityNotificationsEnabled } from 'components/interfaces/App/FeaturePreview/FeaturePreviewContext'
+import { TEMPLATES_SCHEMAS } from 'components/interfaces/Auth/AuthTemplatesValidation'
+import TemplateEditor from 'components/interfaces/Auth/EmailTemplates/TemplateEditor'
+import AuthLayout from 'components/layouts/AuthLayout/AuthLayout'
+import DefaultLayout from 'components/layouts/DefaultLayout'
+import { PageLayout } from 'components/layouts/PageLayout/PageLayout'
+import { ScaffoldContainer, ScaffoldSection } from 'components/layouts/Scaffold'
+import { DocsButton } from 'components/ui/DocsButton'
+import NoPermission from 'components/ui/NoPermission'
+import { useAsyncCheckPermissions } from 'hooks/misc/useCheckPermissions'
+import { DOCS_URL } from 'lib/constants'
+import { useRouter } from 'next/router'
+import { useEffect } from 'react'
+import type { NextPageWithLayout } from 'types'
+import { Card } from 'ui'
+import { GenericSkeletonLoader } from 'ui-patterns'
+
+const TemplatePage: NextPageWithLayout = () => {
+  return <RedirectToTemplates />
+}
+
+const RedirectToTemplates = () => {
+  const router = useRouter()
+  const { templateId, ref } = router.query
+  const isSecurityNotificationsEnabled = useIsSecurityNotificationsEnabled()
+
+  const { can: canReadAuthSettings, isSuccess: isPermissionsLoaded } = useAsyncCheckPermissions(
+    PermissionAction.READ,
+    'custom_config_gotrue'
+  )
+
+  useEffect(() => {
+    if (isPermissionsLoaded && !isSecurityNotificationsEnabled) {
+      router.replace(`/project/${ref}/auth/templates/`)
+    }
+  }, [isPermissionsLoaded, isSecurityNotificationsEnabled, ref, router])
+
+  if (isPermissionsLoaded && !canReadAuthSettings) {
+    return <NoPermission isFullPage resourceText="access your project's email settings" />
+  }
+
+  if (!isSecurityNotificationsEnabled) {
+    return null
+  }
+
+  // Convert templateId slug back to template ID for lookup
+  const templateIdFromSlug = (slug: string) => {
+    const template = TEMPLATES_SCHEMAS.find((template) => {
+      const templateSlug = template.title.trim().replace(/\s+/g, '-').toLowerCase()
+      return templateSlug === slug
+    })
+    return template?.id
+  }
```

Comment:
As above: we need to do this to properly look up e.g. `MAGIC_LINK` from `magic-link`.